client/src/middlewares/messageMiddleware.js

```diff
@@ -0,0 +1,34 @@
+import { webSocketActions, messageActions, userActions } from '../store/actions';
+
+const messageMiddleware = store => next => action => {
+    if (action.type === webSocketActions.types.received) {
+        const parsedMessage = JSON.parse(action.payload);
+        const userId = store.getState().user.id;
+        const actionToDispatch = getActionToDispatch(parsedMessage, userId);
+        if (actionToDispatch) {
+            return store.dispatch(actionToDispatch);
+        }
+    }
+    return next(action);
+};
+
+const getActionToDispatch = (parsedMessage, userId) => {
+    switch (parsedMessage.from.id) {
+        case 'server':
+            return parseServerMessage(parsedMessage);
+        case userId:
+            return;
+        default:
+            return messageActions.addMessages([parsedMessage]);
```

Comment:
default ? why not manage your message types

Reply:
at the moment a message can come from the server, from me (echoed backed from the server) and the rest are message to display.
in the future the default will be replcaed with something more complicated that parse the message to the right place